packages/material-ui/src/InputBase/InputBase.js

```diff
@@ -483,12 +493,7 @@ const InputBase = React.forwardRef(function InputBase(inProps, ref) {
 
   return (
     <React.Fragment>
-      <GlobalStyles
-        styles={{
-          '@keyframes mui-auto-fill': {},
-          '@keyframes mui-auto-fill-cancel': {},
-        }}
-      />
+      <InputGlobalStyles />
```

Comment:
Was this not subject to hoisting of static elements and therefore automatic memoization by React?

If it wasn't (maybe the inline object is the culprit) I would just hoist it manually. I think this should work as well without the need for an additional component:

```diff
-<InputGlobalStyles />
+{inputGlobalStyles}
```

Reply:
Would the `inputGlobalStyles` in this case be:
```diff
-const InputGlobalStyles = React.memo(() => (
+const inputGlobalStyles = (
  <GlobalStyles
    styles={{
      '@keyframes mui-auto-fill': { from: { display: 'block' } },
      '@keyframes mui-auto-fill-cancel': { from: { display: 'block' } },
    }}
  />
-));
+);
```

Let me try this locally

Reply:
Yep, it works. I've pushed the changes.